tests/sentry/incidents/endpoints/test_organization_alert_rule_index.py

```diff
@@ -904,3 +909,78 @@ def test_name_filter(self):
+            resolve_threshold=10,
+            threshold_period=1,
+        )
+        another_alert_rule_warning = self.create_alert_rule(
+            organization=self.org,
+            projects=[self.project],
+            name="another warning rule",
+            query="",
+            aggregate="count()",
+            time_window=1,
+            threshold_type=AlertRuleThresholdType.ABOVE,
+            resolve_threshold=10,
+            threshold_period=1,
+        )
+        alert_rule_warning = self.create_alert_rule(
+            organization=self.org,
+            projects=[self.project],
+            name="warning rule",
+            query="",
+            aggregate="count()",
+            time_window=1,
+            threshold_type=AlertRuleThresholdType.ABOVE,
+            resolve_threshold=10,
+            threshold_period=1,
+        )
+        trigger = self.create_alert_rule_trigger(alert_rule_critical, "hi", 100)
+        trigger2 = self.create_alert_rule_trigger(alert_rule_critical, "bye", 50)
+
+        trigger3 = self.create_alert_rule_trigger(alert_rule_warning, "meow", 200)
+
+        self.create_incident(status=2, alert_rule=alert_rule_critical)
+        warning_incident = self.create_incident(status=10, alert_rule=alert_rule_critical)
+        self.create_incident(status=10, alert_rule=alert_rule_warning)
+        self.create_incident(status=10, alert_rule=another_alert_rule_warning)
+        crit_incident = self.create_incident(status=20, alert_rule=alert_rule_critical)
+        IncidentTrigger.objects.create(
+            incident=crit_incident, alert_rule_trigger=trigger, status=TriggerStatus.RESOLVED.value
+        )
+        IncidentTrigger.objects.create(
+            incident=crit_incident, alert_rule_trigger=trigger2, status=TriggerStatus.ACTIVE.value
+        )
+        IncidentTrigger.objects.create(
+            incident=warning_incident,
+            alert_rule_trigger=trigger3,
+            status=TriggerStatus.ACTIVE.value,
+        )
+        with self.feature(["organizations:incidents", "organizations:performance-view"]):
+            request_data = {
+                "per_page": "10",
+                "project": [self.project.id, self.project2.id],
+                "sort": ["incident_status", "date_triggered"],
+            }
+            response = self.client.get(
+                path=self.combined_rules_url, data=request_data, content_type="application/json"
+            )
+        assert response.status_code == 200, response.content
+        result = json.loads(response.content)
+        assert len(result) == 7
+        # Assert critical rule is first, warnings are next (sorted by triggered date), and issue rules are last.
+        assert result[0]["id"] == str(alert_rule_critical.id)
+        assert result[2]["id"] == str(alert_rule_warning.id)
+        assert result[1]["id"] == str(another_alert_rule_warning.id)
+        assert result[6]["id"] == str(self.issue_rule.id)
```

Comment:
It might be better to just do
`[r['id'] for r in result] == [str(alert_rule_critical.id), ...]`. This ends up giving us better error messages than the individual rows.